src/js/components/DataTable/DataTable.js

```diff
@@ -219,62 +223,35 @@ const DataTable = ({
     console.warn('DataTable cannot combine "size" and "resizeble".');
   }
 
+  const [page, showItem] = normalizeShow(showProp, 'dataTable');
+  const [setPage, currentItems, currentPage] = usePagination({
+    data: adjustedData,
+    paginationProps: { page, showItem, step, ...paginate },
+  });
+
   return (
-    <StyledDataTable fillProp={fill} {...rest}>
-      <Header
-        ref={headerRef}
-        background={normalizeProp(background, 'header')}
-        border={normalizeProp(border, 'header')}
-        columns={columns}
-        data={adjustedData}
-        fill={fill}
-        filtering={filtering}
-        filters={filters}
-        groups={groups}
-        groupState={groupState}
-        pad={normalizeProp(pad, 'header')}
-        pin={pin === true || pin === 'header'}
-        selected={selected}
-        size={size}
-        sort={sort}
-        widths={widths}
-        onFiltering={onFiltering}
-        onFilter={onFilter}
-        onResize={resizeable ? onResize : undefined}
-        onSelect={
-          onSelect
-            ? nextSelected => {
-                setSelected(nextSelected);
-                if (onSelect) onSelect(nextSelected);
-              }
-            : undefined
-        }
-        onSort={sortable || sortProp || onSortProp ? onSort : undefined}
-        onToggle={onToggleGroups}
-        primaryProperty={primaryProperty}
-      />
-      {groups ? (
-        <GroupedBody
-          background={normalizeProp(background, 'body')}
-          border={normalizeProp(border, 'body')}
+    <>
```

Comment:
I think when we have `paginate` we should wrap the table in a Box that contains the table and the pagination, that will provide a place to affect the layout, such as gap and alignment.

Reply:
Good idea. I like that. From a props / theming perspective we can discuss what we'd want the naming structure to be.

Reply:
Added `dataTable.container` and `StyledContainer`. Inside DataTable, if `paginate` is used, then the DataTable and Pagination cmponents are wrapped in `StyledContainer`, otherwise it behaves as is.